tests/function-controller/testsuite/teststep/check.go

```diff
@@ -232,3 +237,103 @@ func (t TracingHTTPCheck) assertTracingResponse(response tracingResponse) error
 
 	return nil
 }
+
+const (
+	resourceLabel     = "serverless.kyma-project.io/resource"
+	functionNameLabel = "serverless.kyma-project.io/function-name"
+	manageByLabel     = "serverless.kyma-project.io/managed-by"
+	uuidLabel         = "serverless.kyma-project.io/uuid"
+)
+
+type APIGatewayFunctionCheck struct {
+	name      string
+	fn        *function.Function
+	client    *v1.CoreV1Client
+	namespace string
+	runtime   string
+}
+
+func NewAPIGatewayFunctionCheck(name string, fn *function.Function, coreV1 *v1.CoreV1Client, ns string, rt string) *APIGatewayFunctionCheck {
+	return &APIGatewayFunctionCheck{
+		name:      name,
+		fn:        fn,
+		client:    coreV1,
+		namespace: ns,
+		runtime:   rt,
+	}
+}
+
+func (d APIGatewayFunctionCheck) Name() string {
+	return d.name
+}
+
+func (d APIGatewayFunctionCheck) Run() error {
+
+	svc, err := d.client.Services(d.namespace).Get(context.Background(), d.name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "while trying to get service")
+	}
+
+	pod, err := d.client.Pods(d.namespace).List(context.Background(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=deployment,%s=%s", resourceLabel, functionNameLabel, d.name)})
+	if err != nil {
+		return errors.Wrap(err, "while trying to get pod")
+	}
+
+	err = checkIfRequiredLabelsExists(svc.Spec.Selector, true)
+	if err != nil {
+		return errors.Wrap(err, " in service")
```

Comment:
I think you should write it in such way: `errors.Wrap(err, "while checking the service labels`